chainer/optimizer.py

```diff
@@ -754,13 +806,16 @@ def update(self, lossfun=None, *args, **kwds):
         self.call_hooks('pre')
 
         self.t += 1
-        for param in self.target.params():
-            param.update()
+        if self.is_safe_to_update():
```

Comment:
I get the impression from this line that the call only "checks if it is safe to update", but it actually does a lot more. E.g. `update_loss_scale` depends on this method being called before. Is it difficult to simplify the logic and decouple these methods?